tools/update_package_source_hashes.rb

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env ruby
```

Comment:
There's a lot of duplicate code between this and its binary equivalent-- would it be possible to combine them into one script?